http/server.ts

```diff
@@ -570,6 +570,13 @@ export async function serve(
   return await server.serve(listener);
 }
 
+/** Serve HTTP requests on port 8000 */
+export async function serve(handler: Handler): Promise<void> {
+  const addr = ":8000";
```

Comment:
@cmorten 
> If we start introducing Deploy specific configuration/concepts/defaults into core std exports then we start to move away from that ideal

The port 8000 is not particularly Deploy specific number. Deploy actually doesn't use the port number. From Deploy's perspective, it's convenient to have ability to 'omit' particular port number (because it's not used).

I guess Ryan suggested 8000 for the default because it's somewhat common port number for local development (for example, python's http.server module seems using 8000). I think I often see some tools have default port number of 3000, 8000, 8080, etc. Either of them are fine and convenient for Deploy's users.

> the initial interfaces were applied verbatim from the RFC #1034 to get us off the ground, and married up closely to the go std lib impl. and therefore seemed fine for a first pass. If use in anger reveals nicer more compat interface then awesome, let's do it!

👍 

> in the case of partial config - do you error, or fallback to non-TLS etc.

I'm in favor of throwing when the partial TLS config is given.

Reply:
> The port 8000 is not particularly Deploy specific number.

Ah apologies, lack of familiarity with deploy and poor assumption making on my part 😅 Scratch that then!